src/xdebugConnection.ts

```diff
@@ -752,21 +752,25 @@ export class Connection extends DbgpConnection {
 
     /** sends a run command */
     public async sendRunCommand(): Promise<StatusResponse> {
+        this.emit('continue-event');
```

Comment:
"ContinueEvent" is a VS Code concept, but this class should have no knowledge of VS Code, only of XDebug. I would prefer if callers of these `send*Command()` functions in `PhpDebugSession` sent the `ContinuedEvent` to VS Code, without this layer of event indirection.

Reply:
@felixfbecker I agree, at the very least it should not emit something that only means something to vscode.

So, when you make any command that executes 1 or more lines of PHP code, the xdebug connection will not respond until the code is done executing, it is essentially locked up.  Given that, maybe it does make sense to have the logic in `xdebugConnection`?  Just change the name to be more appropriate like `code-execution-started`.  Instead of emmitting when vscode status should change to `ContinuedEvent` it is just emmitting that some command was called that will result in PHP code being executed.

I'm fine with doing it either way, changing the event name OR changing the caller to emit the `ContinuedEvent`.

Also if you think this requires more discussion I can move that change to it's own PR?  Since it is not required for the actual deadlock fix, it's more of a UX improvement, I can break it out if you want.